src/controllers/accounts/helpers.js

```diff
@@ -149,7 +149,7 @@ helpers.getUserDataByUserSlug = function (userslug, callerUID, callback) {
 			userData.birthday = validator.escape(String(userData.birthday || ''));
 			userData.moderationNote = validator.escape(String(userData.moderationNote || ''));
 
-			userData['cover:url'] = userData['cover:url'] ? (nconf.get('relative_path') + userData['cover:url']) : require('../../coverPhoto').getDefaultProfileCover(userData.uid);
+			userData['cover:url'] = userData['cover:url'] ? ((/^(http(s?))\:\/\//gi.test(userData['cover:url'])) ? userData['cover:url'] : (nconf.get('relative_path') + userData['cover:url']) ) : require('../../coverPhoto').getDefaultProfileCover(userData.uid);
```

Comment:
Not sure what the memory usage implication is, but I don't think you need to wrap the `s` in a capture group :smile:

`https?:\/\/` is fine.

Reply:
Most of the other places this check is just a `userData['cover:url].startsWith('http')`